rpc/http.go

```diff
@@ -66,6 +66,44 @@ func (hc *httpConn) Close() error {
 	return nil
 }
 
+// HTTPTimeouts represents the configuration params for the HTTP RPC server.
+type HTTPTimeouts struct {
+	// ReadTimeout is the maximum duration for reading the entire
+	// request, including the body.
+	//
+	// Because ReadTimeout does not let Handlers make per-request
+	// decisions on each request body's acceptable deadline or
+	// upload rate, most users will prefer to use
+	// ReadHeaderTimeout. It is valid to use them both.
+	ReadTimeout time.Duration
+
+	// WriteTimeout is the maximum duration before timing out
+	// writes of the response. It is reset whenever a new
+	// request's header is read. Like ReadTimeout, it does not
+	// let Handlers make decisions on a per-request basis.
+	WriteTimeout time.Duration
+
+	// IdleTimeout is the maximum amount of time to wait for the
+	// next request when keep-alives are enabled. If IdleTimeout
+	// is zero, the value of ReadTimeout is used. If both are
+	// zero, ReadHeaderTimeout is used.
+	IdleTimeout time.Duration
+}
+
+const (
+	DefaultHTTPReadTimeout  = 5 * time.Second
+	DefaultHTTPWriteTimeout = 10 * time.Second
+	DefaultHTTPIdleTimeout  = 120 * time.Second
+)
+
+func NewDefaultHTTPTimeouts() HTTPTimeouts {
```

Comment:
It'd be easier to put the defaults in a global struct variable, e.g. DefaultTimeouts. Then you wouldn't need the constants.